### PR TITLE
Model fixes

### PR DESCRIPTION
**Pull request by swatijadhav**
This PR covers following:

- [x] Update model code, use simple conditions
- [x] Use factories to create test records
- [x] Update test cases for models

Changes are still in progress, raising PR to know the comments or any suggestions on the changes.

Thanks

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -110,13 +110,11 @@ class User < ActiveRecord::Base
...
-  scope :active, -> { where('active = ?', true)}
-  scope :by_role, -> (role) { where('role = ?', role) }
+  scope :active, -> { where(active: true)}
+  scope :by_role, -> (role) { where(role: role) }
 
   def set_role_on_invitation_accept
-    if self.role.nil?
-      self.role = "agent"
-    end
+    self.role = self.role.presence || "agent"
```
Comment (CGA1123) [inline app/models/user.rb]: `self.role ||= "agent"`, is even cleaner imo.

Comment (swatijadhav) [reply app/models/user.rb]: the reason to use `self.role.presence` as it will also exclude blank string.

     "" || "agent"
     => ""

Comment (CGA1123) [reply app/models/user.rb]: In theory the roles are predefined and should only ever be one of `[admin, agent, editor, user]`. 

Although I don't see any enforcement of this within the model.
Maybe that should be looked into, good point on the blank string case though.

**Comment by swatijadhav** [comment]
@CGA1123 Changes are now ready for review. Can you please verify and merge. Thanks.

**Comment by CGA1123** [comment]
LGTM!
I don't have the permission to merge unfortunately. Will have to wait for @scott to approve and merge.

@scott, is there a project wide push toward replacing fixtures with `FactoryGirl`?

File: app/models/user.rb (hunk 1 above)
Comment (scott) [reply app/models/user.rb]: I would leave it unrestricted in case someone wants to add their own values.

File: app/models/user.rb [hunk 2]
```diff
@@ -130,15 +128,15 @@ def enable_notifications_for_admin
...
   end
 
   def self.notifiable_on_private
-    User.agents.where(notify_on_private: true).reorder('id asc')
+    self.agents.where(notify_on_private: true).reorder('id asc')
   end
 
   def self.notifiable_on_reply
-    User.agents.where(notify_on_reply: true).reorder('id asc')
+    self.agents.where(notify_on_reply: true).reorder('id asc')
```
Comment (scott) [inline app/models/user.rb]: Is it in the style guide to prefer `self` here?  Also, why not just shorten it to 
```
def self.notifiable_on_public
   agents.where(notify_on_public: true).reorder('id asc')
end
```

**Comment by swatijadhav** [comment]
Changes are done as per the suggestion. Thanks @scott

**Pull request merged**